tests/flytekit/unit/core/test_interface.py

```diff
@@ -320,6 +320,16 @@ def z(a: int, b: str) -> typing.NamedTuple("NT", x_str=str, y_int=int):
     assert typed_interface.outputs.get("y_int").description == "description for y_int"
 
 
+def test_init_interface_with_invalid_parameters():
+    from flytekit.core.interface import Interface
+
+    with pytest.raises(ValueError, match=r"Input must be valid Python identifier:"):
+        _ = Interface({"my.input": int}, {})
+
+    with pytest.raises(ValueError, match=r"Type names and field names must be valid identifiers:"):
```

Comment:
The error message is odd here, but it is the standard error message from `collections.namedtuple` see: https://github.com/flyteorg/flytekit/pull/2538/files#r1657915913  https://github.com/python/cpython/blob/main/Lib/collections/__init__.py#L401-L402